File: Java/src/main/java/OneLang/Test/TestCase/TestCase.java
package OneLang.Test.TestCase;



import java.util.function.Consumer;

public class TestCase {
    public String name;
    public Consumer<String> action;
    
    public TestCase(String name, Consumer<String> action)
    {
        this.name = name;
        this.action = action;
    }
}

File: CSharp/Test/TestCase.cs
using System.Threading.Tasks;
using System;

namespace Test
{
    public interface ITestCollection {
        string name { get; set; }
        
        TestCase[] getTestCases();
    }
    
    public class TestCase
    {
        public string name;
        public Func<string, Task> action;
        
        public TestCase(string name, Func<string, Task> action)
        {
            this.name = name;
            this.action = action;
        }
    }
    
    public class SyncTestCase : TestCase
    {
        public Action<string> syncAction;
        
        public SyncTestCase(string name, Action<string> syncAction): base(name, null)
        {
            this.syncAction = syncAction;
            this.action = artifactDir => this.execute(artifactDir);
        }
        
        public async Task execute(string artifactDir)
        {
            this.syncAction(artifactDir);
        }
    }
}